src/db-connection.ts

```diff
@@ -0,0 +1,77 @@
+import { config } from 'node-config-ts'
+import type { Knex } from 'knex'
+import knex from 'knex'
+import snakeCase from 'lodash.snakecase'
+import camelCase from 'lodash.camelcase'
+import { logger } from './logger/index.js'
+
+const KNEX_TABLES = ['knex_migrations', 'knex_migrations_lock']
+
+function toCamelCase(value) {
+  if (Array.isArray(value)) {
+    return value.map(toCamelCase)
+  }
+
+  if (typeof value === 'object') {
+    return Object.fromEntries(Object.entries(value).map(([key, value]) => [camelCase(key), value]))
+  }
+
+  if (typeof value === 'string') {
+    return camelCase(value)
+  }
+
+  return value
+}
+
+async function runMigrations(connection: Knex) {
+  const [_, pendingMigrations] = await connection.migrate.list()
+  if (pendingMigrations.length > 0) {
+    logger.imp(`Running ${pendingMigrations.length} migrations`)
+    await connection.migrate.latest()
+  }
+}
+
+export async function createDbConnection() {
+  const knexConfig: Knex.Config = {
+    ...config.db,
+    wrapIdentifier: (value, origWrap): string => origWrap(snakeCase(value)),
+    postProcessResponse: (result) => toCamelCase(result),
```

Comment:
Can you add comments explaining what this is doing and why it's needed?

Reply:
sure but just to explain here this is used to convert identifiers between camel case and snake case instead of having out code written with bot. Its only a preference to keep things consistent within the code but if you and anyone else finds it unnecessary I can remove it.